rocAL/rocAL_pybind/rocal_pybind.cpp

```diff
@@ -304,10 +297,236 @@ namespace rocal{
+            .def_readwrite("transfer_time", &TimingInfo::transfer_time);
+        py::class_<rocalTensor>(m, "rocalTensor")
+                .def(
+                "batch_height",
+                [](rocalTensor &output_tensor)
+                {
+                    return output_tensor.info().max_shape().at(1);
+                },
+                R"code(
+                Returns a tensor buffer's height.
+                )code"
+            )
+            .def(
+                "batch_width",
+                [](rocalTensor &output_tensor)
+                {
+                    return output_tensor.info().max_shape().at(0);
+                },
+                R"code(
+                Returns a tensor buffer's width.
+                )code"
+            )
+            .def(
+                "batch_size",
+                [](rocalTensor &output_tensor)
+                {
+                    return output_tensor.info().dims().at(0);
+                },
+                R"code(
+                Returns a tensor batch size.
+                )code"
+            )
+            .def(
+                "color_format",
+                [](rocalTensor &output_tensor)
+                {
+                    switch(output_tensor.info().color_format())
+                    {
+                        case RocalColorFormat::RGB24:
+                            return 3;
+                        case RocalColorFormat::BGR24:
+                            return 3;
+                        case RocalColorFormat::RGB_PLANAR:
+                            return 3;
+                        case RocalColorFormat::U8:
+                            return 1;
+                    }
+                },
+                R"code(
+                Returns a tensor batch size.
+                )code"
+            )
+            .def("layout", [](rocalTensor &output_tensor)
+            {
+                switch(output_tensor.info().layout()) {
+                    case RocalTensorlayout::NHWC:
+                        return "NHWC";
+                    case RocalTensorlayout::NCHW:
+                        return "NCHW";
+                }
+            },
+                R"code(
+                Returns layout of tensor.
+                )code"
+            )
+            .def("dtype", [](rocalTensor &output_tensor)
+            {
+                switch(output_tensor.info().data_type()) {
+                    case RocalTensorDataType::FP32:
+                        return "float32";
+                    case RocalTensorDataType::UINT8:
+                        return "uint8";
+                    case RocalTensorDataType::FP16:
+                        return "float16";
+                }
+            },
+                R"code(
+                Returns dtype of tensor.
+                )code"
+            )
+            .def(
+            "copy_data", [](rocalTensor &output_tensor, py::object p)
+            {
+                auto ptr = ctypes_void_ptr(p);
```

Comment:
We are doing a ctypes_void_ptr here and whenever we are calling this API also we are doing a ctypes void ptr cast, Is both necessary or can we remove any 1

Reply:
Both are necessary, DALI does it in a similar way